File: mc/rcmd/src/main/java/calebxzhou/rdi/mc/rcmd/Rcmd.java
package calebxzhou.rdi.mc.rcmd;

public final class Rcmd {
    public static final char PREFIX = '\\';

    private Rcmd() {
    }

    public static boolean isRcmd(String input) {
        return input != null && input.startsWith(String.valueOf(PREFIX));
    }

    public static String stripPrefix(String input) {
        if (!isRcmd(input)) {
            return input;
        }
        return input.substring(1);
    }
}

File: mc/rcmd/src/main/kotlin/calebxzhou/rdi/mc/rcmd/Rcmd.kt
package calebxzhou.rdi.mc.rcmd

object Rcmd {
    const val PREFIX: Char = '\\'

    @JvmStatic
    fun isRcmd(input: String) = input.startsWith(PREFIX)

    @JvmStatic
    fun stripPrefix(input: String) = if (isRcmd(input)) input.substring(1) else input
}
